### PR TITLE
add docs/bundles/helm.md

### PR DESCRIPTION
**Pull request by akihikokuroda**
Part of #522 

Signed-off-by: akihikokuroda <akuroda@us.ibm.com>

### CODE REVIEW

File: docs/bundles/helm.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+Supported source types for a helm bundle currently include the following:
+
+* A container image
+* A directory in a git repository
+* A [http](../sources/http.md)
+* An [upload](../uploading-bundles.md)
+
+Additional source types, such as a local volume are on the roadmap. These source types
+all present the same content, a directory containing a helm chart, in a different ways.
```
Comment (joelanford) [inline docs/bundles/helm.md:27]: I think this is more appropriate for the helm provisioner docs.

The helm+v3 format itself doesn't require a specific set of "transport" formats. If someone else wants to implement a different helm provisioner with a different set of source types, that has no impact on the helm bundle format.

File: docs/bundles/helm.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+content embedded in the bundle. The bundle can be used as the underlying `spec.source` for
+a [Bundle](https://github.com/operator-framework/rukpak#bundle) resource.
+
+A helm bundle is a [helm chart](https://helm.sh/docs/chart_template_guide/getting_started/#charts).
+The helm chart can be created by [helm create](https://helm.sh/docs/helm/helm_create/).
+
+The currently implemented helm bundle format is the `helm+v3` format. The name of the bundle format, `helm+v3`
+combines the type of bundle (helm) with the current helm format version (v3).
+The [helm bundle provisioner](https://github.com/operator-framework/rukpak/blob/main/internal/provisioner/helm/README.md)
+is able to source `helm+v3` bundles and install them onto a Kubernetes cluster.
```
Comment (joelanford) [inline docs/bundles/helm.md:17]: I think conceptually, we shouldn't be talking about what's implemented or not in this doc. In my mind, this doc is simply "here the spec of a helm+v3 bundle; if you're a helm provisioner and you accept a helm+v3 bundle, you must implement unpacking/validation logic according to this spec."

File: docs/bundles/helm.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+│   ├── _helpers.tpl
+│   ├── deployment.yaml
+│   ├── service.yaml
+│   └── serviceaccount.yaml
+└── values.yaml
+```
+
+## Technical Details
+
+* Only one chart should be in the bundle.
```
Comment (joelanford) [inline docs/bundles/helm.md:49]: IMO, this is a slightly strange statement to make and could be confusing (e.g. "can my helm+v3 bundle specify dependencies?). I think we should define the helm+v3 bundle as a single helm chart. So by definition, a helm+v3 bundle _is_ a helm chart.

File: docs/bundles/helm.md (hunk 1 above)
Comment (akihikokuroda) [reply docs/bundles/helm.md:27]: OK.  I'll move to the helm provisioner doc.

File: docs/bundles/helm.md (hunk 2 above)
Comment (akihikokuroda) [reply docs/bundles/helm.md:17]: OK.  I take this out.

File: docs/bundles/helm.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+The current helm bundle format name is the `helm+v3` that
+combines the type of bundle (helm) with the current helm format version (v3).
+
+## Example
+
+For example, below is an example of a file tree in a "helm+v3" bundle.
+
+```tree
+hello-world
```
Comment (joelanford) [inline docs/bundles/helm.md:22]: As of https://github.com/operator-framework/rukpak/pull/516, we support helm chart bundles with or without a root chart directory. So at least the helm provisioner supports both of these:
```
<bundleRoot>
└── nginx
    ├── Chart.yaml
    ├── charts
    ├── templates
    │   ├── NOTES.txt
    │   ├── _helpers.tpl
    │   ├── deployment.yaml
    │   ├── hpa.yaml
    │   ├── ingress.yaml
    │   ├── service.yaml
    │   ├── serviceaccount.yaml
    │   └── tests
    │       └── test-connection.yaml
    └── values.yaml
```

and

```
<bundleRoot>
├── Chart.yaml
├── charts
├── templates
│   ├── NOTES.txt
│   ├── _helpers.tpl
│   ├── deployment.yaml
│   ├── hpa.yaml
│   ├── ingress.yaml
│   ├── service.yaml
│   ├── serviceaccount.yaml
│   └── tests
│       └── test-connection.yaml
└── values.yaml
```

IMO, we should specify in the spec that implementations MUST support both of these.

Comment (akihikokuroda) [reply docs/bundles/helm.md:22]: The contents of the tgz file generated by the `helm package` command has the first format.  So the current implementation of the helm provisioner is only supporting the first format.  We can create a new issue to support the second tree and update the docs/bundle/helm.md.  Is it OK?

Comment (joelanford) [reply docs/bundles/helm.md:22]: I added support for the second in that PR I linked so that `rukpakctl run my-bd ./<chartDir>` would work.

Comment (akihikokuroda) [reply docs/bundles/helm.md:22]: I see.  I'll add the second tree.

**Comment by akihikokuroda** [comment]
Put into #552

**Pull request closed without merging**